account_analytic_default_tag/models/account_analytic_default.py

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class AccountAnalyticDefault(models.Model):
+
+    _inherit = "account.analytic.default"
+
+    analytic_tag_ids = fields.Many2many('account.analytic.tag',
+                                        string="Analytic tags")
+
+    @api.multi
+    @api.constrains('analytic_id', 'analytic_tag_ids')
+    def _check_required(self):
+        # As at least one field of analytic_id and tag_ids is required we throw
+        # a validation error should both fields be empty
+        for default in self:
+            if not default.analytic_id and not default.analytic_tag_ids:
+                raise ValidationError(_(
+                    'At least one Analytic Account or Analytic Tag has to be '
+                    'defined.'))
+
+    @api.model
+    def create(self, vals):
+        if 'analytic_id' not in vals and 'analytic_tag_ids' not in vals:
+            raise ValidationError(_(
+                'At least one Analytic Account or Analytic Tag has to be '
+                'defined.'))
```

Comment:
With constraints you defined `_check_required`, do we need to do the check again here?

Reply:
IIRC the check with api.constrains, won't happen if those fields are not used in the record creation, but I should probably have added a comment to explicit this.

Reply:
Thanks for explanations :)